build-system/babel-plugins/babel-plugin-transform-amp-asserts/index.js

```diff
@@ -94,20 +94,16 @@ module.exports = function(babel) {
       }
     }
 
-    path.replaceWith(t.parenthesizedExpression(arg));
-
     if (type) {
+      path.replaceWith(t.parenthesizedExpression(arg));
       // If it starts with a capital, make the type non-nullable.
```

Comment:
What about cases like `string` where they're sometimes non-nullable? For example:

https://github.com/ampproject/amphtml/blob/54eec5e160992323d3c3c07bd284cbfbc5ede212/third_party/subscriptions-project/swg.js#L6812-L6817

Or is this incorrect?

Reply:
The difference is nullable by default. For Objects, `{Object}` and `{?Object}` are equivalent, and have to be changed to `{!Object}` to prove it can't be null. For primitives, `{string}` and `{!string}` are equivalent, and we don't need to do anything to prove it's not null.